### PR TITLE
bugfix: StreamResponseFilter of data filter framework can't work if size of one object is over 32KB

### PR DESCRIPTION
**Pull request by rambohe-ch**
/kind bug

#### What this PR does / why we need it:
1. bugfix: StreamResponseFilter of data filter framework can't work if size of one object is over 32KB
2. improve data filter framework
  - define ObjectHandler interface for filtering Runtime.Object, and other common routines are removed to FilterReadCloser
  - add SerializerManager as common parameter for all filters.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/yurthub/filter/filter.go [hunk 1]
```diff
@@ -115,89 +116,151 @@ func (fis FilterInitializers) Initialize(ins Runner) error {
...
+	return frc.rc.Close()
+}
+
+func (frc *filterReadCloser) ObjectResponseFilter(rc io.ReadCloser) (*bytes.Buffer, error) {
+	var buf bytes.Buffer
+	_, err := buf.ReadFrom(rc)
+	if err != nil {
+		return &buf, err
+	}
+	listObj, err := frc.serializer.Decode(buf.Bytes())
```
Comment (Congrool) [inline pkg/yurthub/filter/filter.go:208]: It seems that we do not only handle listobject but also standalone object. So, it maybe unsuitable to use the variable name as `listObj`.

File: pkg/yurthub/filter/filter.go [hunk 2]
```diff
@@ -115,89 +116,151 @@ func (fis FilterInitializers) Initialize(ins Runner) error {
...
+
+	newData, err := frc.serializer.Encode(filteredObj)
+	return bytes.NewBuffer(newData), err
+}
 
-			return n, nil
+func (frc *filterReadCloser) StreamResponseFilter(rc io.ReadCloser, ch chan *bytes.Buffer) error {
+	defer func() {
+		close(ch)
+	}()
```
[the remark this answers is not in the document]
Comment (rambohe-ch) [reply pkg/yurthub/filter/filter.go]: fixed

File: pkg/yurthub/filter/filter.go (hunk 1 above)
Comment (rambohe-ch) [reply pkg/yurthub/filter/filter.go]: listObj --> obj. 
fixed

**Comment by rambohe-ch** [comment]
@Congrool PTAL

**Pull request merged**